protocol/parser/parser.go

```diff
@@ -122,105 +123,120 @@ func parseInputFromParamsWithGenericParsers(rpcInput RPCInput, genericParsers []
-		}
-
-		resString, ok := result[0].(string)
-		if !ok {
-			utils.LavaFormatDebug("ParseBlockFromParams - result[0].(string) - type assertion failed", utils.LogAttr("result[0]", result[0]))
-			return spectypes.NOT_APPLICABLE
-		}
-		parsedBlock, err := rpcInput.ParseBlock(resString)
-		if err != nil {
-			if blockParser.DefaultValue != "" {
-				utils.LavaFormatDebug("Failed parsing block from string, assuming default value",
-					utils.LogAttr("params", rpcInput.GetParams()),
-					utils.LogAttr("failed_parsed_value", resString),
-					utils.LogAttr("default_value", blockParser.DefaultValue),
+			parsedBlock, err = rpcInput.ParseBlock(defaultValue)
+			if err != nil {
+				utils.LavaFormatError("Failed parsing default value, setting to NOT_APPLICABLE", err,
+					utils.LogAttr("default_value", defaultValue),
 				)
-				parsedBlock, err = rpcInput.ParseBlock(blockParser.DefaultValue)
-				if err != nil {
-					utils.LavaFormatError("Failed parsing default value, setting to NOT_APPLICABLE", err,
-						utils.LogAttr("default_value", blockParser.DefaultValue),
-					)
-					return spectypes.NOT_APPLICABLE
-				}
-			} else {
-				return spectypes.NOT_APPLICABLE
+				parsedBlock = spectypes.NOT_APPLICABLE
 			}
+		} else {
+			parsedBlock = spectypes.NOT_APPLICABLE
 		}
-		return parsedBlock
-	}()
-
-	return parsedBlockInfo
+	}
+	parsedInput.SetBlock(parsedBlock)
 }
 
-// This returns the parsed response without decoding
-func ParseFromReply(rpcInput RPCInput, blockParser spectypes.BlockParser) (string, error) {
-	result, err := parse(rpcInput, blockParser, PARSE_RESULT)
+func parseInputWithLegacyBlockParser(rpcInput RPCInput, blockParser spectypes.BlockParser, source int) (string, error) {
+	result, err := legacyParse(rpcInput, blockParser, source)
 	if err != nil || result == nil {
-		utils.LavaFormatDebug("ParseBlockFromParams - parse failed",
+		return "", utils.LavaFormatDebug("blockParsing - parse failed",
 			utils.LogAttr("error", err),
 			utils.LogAttr("result", result),
 			utils.LogAttr("blockParser", blockParser),
 			utils.LogAttr("rpcInput", rpcInput),
 		)
-		return "", err
 	}
 
-	response, ok := result[spectypes.DEFAULT_PARSED_RESULT_INDEX].(string)
+	resString, ok := result[spectypes.DEFAULT_PARSED_RESULT_INDEX].(string)
 	if !ok {
-		return "", utils.LavaFormatError("Failed to Convert blockData[spectypes.DEFAULT_PARSED_RESULT_INDEX].(string)", nil, utils.Attribute{Key: "blockData", Value: response[spectypes.DEFAULT_PARSED_RESULT_INDEX]})
+		return "", utils.LavaFormatDebug("blockParsing - result[0].(string) - type assertion failed", utils.LogAttr("result[0]", result[0]))
 	}
 
-	if strings.Contains(response, "\"") {
-		responseUnquoted, err := strconv.Unquote(response)
-		if err != nil {
-			return response, nil
-		}
-		return responseUnquoted, nil
+	return resString, nil
+}
+
+func parseBlock(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser, source int) *ParsedInput {
```

Comment:
comment what it does

Reply:
Added comment in 79e54af9705b33196af60d4bf41aed161b0b344a

Reply:
Added comment in 79e54af9705b33196af60d4bf41aed161b0b344a